### ISSUE TITLE
Listing grid - Order by "if post has been promoted" otherwise by relevance

### ISSUE THREAD

**Issue by mattkub** [CrocoBlock/suggestions#2982]
I would like to order my posts by 2 criteria. First posts would be promoted (posts has taxonomy "promoted" --> "top") and other posts (post taxonomy "promoted" is empty). 

I would like to display promoted posts always on the top of the grid and other posts will be ordered by relevance. I still cant figure out how can i make it. 

If you got any idea "out of the box" how can i solve this issue, it will help me too.

**Comment by mahroch** [comment]
I asked the same  here https://github.com/Crocoblock/suggestions/issues/2896 :)

**Comment by mattkub** [comment]
@mahroch i am not sure if we facing to same issue. I got bunch of posts that can have 2 type of status "promoted" and "free" promoted should be always shown first and free should be ordered by relevance after promoted onece :) but thanks i will watchout your issue too :)

**Comment by mahroch** [comment]
yes , it is the same, at least as I understand your suggestion. For me, if we call it promoted, or sticky is the same ... ordering the other posts except sticky/promoted/featured ones can be the same as now - that means there is possibility of relevance if I remember correctly ...

**Comment by mattkub** [comment]
@mahroch I probably made it. i made meta query for "promotion" ==> promotion_clause and meta query for "date" ==> date_clause and then add order_offset for meta clause by promotion_clause. 

I am still testing this solution but you can try to make it work this way
